### DISCUSSION TITLE
Do I need to create supabase policy if I use prisma?

### DISCUSSION THREAD

**Discussion by rajanshresth** [supabase/supabase#14368] [General]
Do I need to create policy if I use prisma?

I don't know SQL but i want to use supabase as I've some experience with prisma...!

**Comment by souvik666** [comment]
If you would like to use Row Level Security (RLS) with Prisma, check out the Prisma Client Extension - Row Level Security example that provides the primitives you could use to build and extend Prisma Client in PostgreSQL.

Also check out useSupabaseRowLevelSecurity Prisma Client extension that supports Supabase RLS and policies written to use Supabase auth.

The example and extension use Prisma Client extensions Preview feature.

Enabling PosgreSQL extensions#
If you would like to use a PostgreSQL extension with Prisma, enable the postgresqlExtensions Preview feature flag in the generator block of your Prisma schema:

```
datasource db {
  provider          = "postgresql"
  url               = env("DATABASE_URL")
  directUrl         = env("DIRECT_URL")
}

generator client {
  provider        = "prisma-client-js"
  previewFeatures = ["postgresqlExtensions"]
}
```

Next, specify the extensions you need in the datasource block:

```
datasource db {
  provider          = "postgresql"
  url               = env("DATABASE_URL")
  directUrl         = env("DIRECT_URL")
  extensions        = [hstore(schema: "myHstoreSchema"), pg_trgm, postgis(version: "2.1")]
}

generator client {
  provider        = "prisma-client-js"
  previewFeatures = ["postgresqlExtensions"]
}
```

please follow this [link ](https://supabase.com/docs/guides/integrations/prisma)for elaborative doc on the same 
To learn more about using Prisma with PostgreSQL extensions, refer to [this page in the Prisma docs](https://www.prisma.io/docs/concepts/components/prisma-schema/postgresql-extensions).

**Reply by rajanshresth** [reply]
**What does these extensions exactly do?**
